### ISSUE TITLE
Help wanted: Update translations

### ISSUE THREAD

**Issue by OzzieIsaacs** [janeczku/calibre-web#606]
Many of the translations are outdated due to the fast evolving of the code. 
Any contribution to the translations is welcome.
Just edit the to the language corresponding ".po" file in the cps/translations folder (up to the line: "#. name for aaa", approx line 1900) and create a PR request. I will include it to calibre Web. 
(Editing this file will not make the text instantly translated in Calibre Web, they have to be compiled first with poedit, pyBabel, or something else...)

**Comment by vigri** [comment]
I can take a look at the translations for German and add Slovak if needed

**Comment by OzzieIsaacs** [comment]
Thanks, the German version is always in good shape. You can add Slovak if you want. 
(Aber wir brauchen keine weitere Übersetzung, um sie zu haben und dann nicht zu pflegen)

**Comment by Ribero** [comment]
I updated the french version. It's a fist try. I will update it more regularly and will try to take into account any suggestion, if any.

**Comment by eddriesen** [comment]
Hi Ozzie! I've finished the Dutch translation (WOW! you did add a lot of functionality!), but I keep bumping into errors in poedit. I have 6 errors (id  17,16,442 and 129). poedit claims these are "double definitions". Is this fixable on my end? Anyhoo, I'll attacht the uncompiled PO.
[messages.po.txt](https://github.com/janeczku/calibre-web/files/2408658/messages.po.txt)

It prevens me from compiling to MO

**Comment by OzzieIsaacs** [comment]
Poedit serms to get confused by the invalid marked strings (starting with #) at the end of the file.

**Comment by eddriesen** [comment]
> Poedit seems to get confused by the invalid marked strings (starting with #) at the end of the file.

Yup! Anyhoo, dove into the deep end and dit a pull request.... Git is a very confusing platform for the uninitiated. ;-)

**Comment by candidodmv** [comment]
how include new language to translate? I want bring pt-BR to the project. Thanks

**Comment by OzzieIsaacs** [comment]
In the root folder of the project is the template file for translations, include the translated texts in that file. You can add a folder with the translated and compiled file (the mo file) in the translation folder parallel to all the other translations and create a pull request. If you are not so familiar with git and  github you could also create a new issue and upload the language file to it, I'll merge it afterwards.

**Comment by hashevskyi** [comment]
Translation into Ukrainian 
[messages.po.txt](https://github.com/janeczku/calibre-web/files/2822304/messages.po.txt)

**Comment by Ribero** [comment]
French version updated.
[messages.po.txt](https://github.com/janeczku/calibre-web/files/2825568/messages.po.txt)

**Comment by floviolleau** [comment]
> 
> 
> French version updated.
> [messages.po.txt](https://github.com/janeczku/calibre-web/files/2825568/messages.po.txt)

I found a mistake `utilisateurs` instead of `utulisateurs`

**Comment by OzzieIsaacs** [comment]
It's merged on the newest commit

**Comment by subdiox** [comment]
Japanese version updated
https://github.com/subdiox/calibre-web/blob/master/cps/translations/ja/LC_MESSAGES/messages.po

**Comment by ElQuimm** [comment]
Italian version updated.
[messages.po.txt](https://github.com/janeczku/calibre-web/files/3139718/messages.po.txt)

**Comment by LoafingBunny** [comment]
> Italian version updated.
> [messages.po.txt](https://github.com/janeczku/calibre-web/files/3139718/messages.po.txt)

Aggghh I was updating it too... my fault. Just a notice: it's always funny to read Source translated as Spring

**Comment by OzzieIsaacs** [comment]
Thanks a lot for the update @ElQuimm. I merged the update.

**Comment by ElQuimm** [comment]
> Thanks a lot for the update @ElQuimm. I merged the update.
Thank you for Calibre-Web :-) @OzzieIsaacs and "ciao" to @LoafingBunny.

**Comment by ElQuimm** [comment]
> 
> 
> > Thanks a lot for the update @ElQuimm. I merged the update.
> > Thank you for Calibre-Web :-) @OzzieIsaacs and "ciao" to @LoafingBunny.

Hello again.
I have revised one more time the italian version. 
[message.po.txt](https://github.com/janeczku/calibre-web/files/3166408/message.po.txt)

**Comment by joshobrien91** [comment]
How can I update/make changes to the English translations, can't find any English translation option to change the default messaging. 

Many of the error message or user prompts are more geared to a technical user, than a general user, which most of my users will be. Like to soften the messages throughout.

**Comment by OzzieIsaacs** [comment]
In the root folder you'll find the file message.pot. In this file you find all untranslated strings. Above every string you see a line in the format "# path/filename.ending:code line". Like this one:
```
#: cps/about.py:78
msgid "Statistics"
msgstr 
```
No go to the code line in the said file and change the text right in the code. The consequence is, you will see the new text after a restart of calibre-web, and you break the translation of this text for anybody else (we have to re translate afterwards). If you create a pull request I can merge your changes.

**Comment by joshobrien91** [comment]
@OzzieIsaacs I tried that, however the text remains unchanged. 
This is when I update the text in the msgid in this example editing Statistics. 

```
#: cps/about.py:78
msgid "Statistics"
msgstr 
```

Or do you mean actually go to each individual source file and manually rewrite it there? ie go to 

`cps/about.py` file and alter the source?

Even if I remove the message.pot file then reboot it doesn't cause a fatal error or show any different text, text remains unchanged. 

I have since updated the source code with all the required language changes for my environment, however just for ease of future updates, it'd be preferable if English was also a reference translation file, like other languages, to enable easy changes to all the English messaging throughout, without altering the source.

**Comment by OzzieIsaacs** [comment]
The message.pot is only a collections of all texts to translate into different languages. You have to change the sourcefile (e.g. in cps/about.py), because the texts to translated are extracted from the source, and it's not done by referencing to text_id xy and looking for this id in a text catalog. 
If you provide a pull request with your changes, I can generate a new pot files (it's done automatically). If the text to translate is changed from e.g. "Crazy text" to "Not so crazy text" this text has to be changed for all languages because the meaning has changed.

**Comment by minakmostoles** [comment]
Today I did update and revision of the Spanish translation file. Thank you for this project is great project

**Comment by RPSouto** [comment]
Do you need a Portuguese version?

**Comment by OzzieIsaacs** [comment]
The question is more If you want to have one? 
If you create a translation I will add it to the project, no question.

**Comment by adocampo** [comment]
Reviewed and added the missing sentences on the spanish translation.

I noticed, but, there are a missing translation (My Profile) I didn't see in the .po file. I don't know if is missing globally or just in spanish.

![image](https://user-images.githubusercontent.com/2266612/63766730-1e5ff700-c8cc-11e9-97e0-d0acd77027ea.png)

**Comment by OzzieIsaacs** [comment]
The caliblur theme creates some texts, this one‘s are not all translatable (I missed some of them)

**Comment by adocampo** [comment]
No problem, if I notice another missing label, I will notify :)

**Comment by svalavuo** [comment]
Initial translation to Finnish :-)

[messages.mo.txt](https://github.com/janeczku/calibre-web/files/3711595/messages.mo.txt)
[messages.po.txt](https://github.com/janeczku/calibre-web/files/3711596/messages.po.txt)

**Comment by OzzieIsaacs** [comment]
Sorry for the long response, somehow I forgot your post. Now it's merged. I tried to reproduce your problem (you are referring to the load metadata button in the edit book section?), but couldn't manage it. If the problem still persists, please open a new issue, post your OS version, docker provider (if any), unnormaly hardware (nas, raspberr pi, ...) the relevant contents of the logfile, python version and under which circumstances which problem exactly occurs.

**Comment by eddriesen** [comment]
HeyHey,

I'm working on an update for the Dutch files;However I would like to add some handy pointers for anyone who's interested;

- When a sentence starts with a Capital, translate with a capital
- Same for punctuation marks, when a sentence ends with one, best that you do to.
-Some translations have a white space after the final sentence/punctuation; translate with ending white space.
- Keep context and tone! Sometimes trying to translate something to literal sounds unnatural in you native language. You should adopt the phrasing WITHOUT changing context, and without changing the tone (formal <-> familiar)

Anyhoo, just some tips I personally had to learn the hard way (full internet wrath ;-) ). Especially the last point is a hard one. 

Poedit (thank god!) filters out a lot of these errors 

Also, the file I pulled had already some syntax errors as I described n posts above this one. Anyone knows if this is  Poedit's fault, or just a general syntax error?

Any thoughts? 

Disclaimer: I to am guilty of constructing wonky sentences. I am by al means no grammarn#z#.

![Schermafdruk_2020-03-06_19-56-56](https://user-images.githubusercontent.com/29610617/76113978-70a17380-5fe5-11ea-8762-c95c24368311.png)

**Comment by OzzieIsaacs** [comment]
There had been  some double strings in the file at the end, if deleted them. I think this happened during one of the merges. If this happens again, feel free to delete the strings (starting with #~)
I'm not updating all translation files manually, I have a script for it. The script has no problem with the double strings, so I never realized it.

**Comment by eddriesen** [comment]
Pushed what I had, should a good improvement and addition. Will do a word-by-word check-up when things settle back (Schools closed for the next 5 weeks and wife in quarantaine, so full hands), but hey! we've got our fancy library to catch up on some reading!

**Comment by dekani** [comment]
Here is, as attached files, a full updated translation for French language with latest changes.
Best regards.
[messages.mo.txt](https://github.com/janeczku/calibre-web/files/4599300/messages.mo.txt)
[messages.po.txt](https://github.com/janeczku/calibre-web/files/4599301/messages.po.txt)

**Comment by OzzieIsaacs** [comment]
Thank you very much, it's merged

**Comment by Knepherbird** [comment]
@OzzieIsaacs quick question regarding changes to strings in the UI and the .po files. There's a setting for "Allow Editing Public Shelfs" that should be "Allow Editing Public Shelves". If that's corrected in, I'm assuming config_view_edit.html and user_edit.html, then the string will need to be updated in messages.pot and all of the current .po files, correct?

**Comment by OzzieIsaacs** [comment]
@Knepherbird you are right, the string has to be changed in all the files you mentioned. Normally I'm having a script for this, but in this case (typo in english text), I'd like to keep the translated texts. So changing manually the relevant strings is the way of doing it.

**Comment by Knepherbird** [comment]
Cool, just submitted a PR

**Comment by dekani** [comment]
Here is an updated translation for French language with latest changes and some fixes. Best regards.
[messages.po.txt](https://github.com/janeczku/calibre-web/files/4738612/messages.po.txt)

**Comment by OzzieIsaacs** [comment]
@dekani Thanks for the translation, it's merged.

**Comment by y12345e** [comment]
Chinese Translations
[Desktop.zip](https://github.com/janeczku/calibre-web/files/4792944/Desktop.zip)

**Comment by OzzieIsaacs** [comment]
Thanks a lot, it's merged to master branch

**Comment by y12345e** [comment]
Chinese Translations V0.6.9
[LC_MESSAGES.zip](https://github.com/janeczku/calibre-web/files/5288432/LC_MESSAGES.zip)

**Comment by OzzieIsaacs** [comment]
@y12345e thanks

**Comment by morabsf** [comment]
Hi, I've just updated the italian translation.
[messages.po.txt](https://github.com/janeczku/calibre-web/files/5771474/messages.po.txt)
:)

**Comment by iAHTOH** [comment]
Hi! I have updated the Russian translation
[messages.zip](https://github.com/janeczku/calibre-web/files/5840973/messages.zip)

**Comment by jonbarbosa** [comment]
Brazilian Portuguese translation
[messages-pt_BR.zip](https://github.com/janeczku/calibre-web/files/6215558/messages-pt_BR.zip)

**Comment by OzzieIsaacs** [comment]
Thank you I got your emails from yesterday and from today, and I also read this thread

**Comment by fqscfqj** [comment]
Chinese translation increment
[messages.po.txt](https://github.com/janeczku/calibre-web/files/6371909/messages.po.txt)

**Comment by OzzieIsaacs** [comment]
Thanks

**Comment by JFernando122** [comment]
I'm currently updating the spanish translation, but i got some questions
1.  above some translated lines theres a `#, fuzzy` comment, i should i change those transalations if i can?
2. Some translations include empty curly braces `{}`, those also go  into the translation?
3. what exactly is "LDAP Member User Filter"?

**Comment by OzzieIsaacs** [comment]
@JFernando122 
1: if the translation should be improved yes
2. yes, please keep the braces
3. for login with a ldap server. It is a filtering expression to find the username (more or less)

**Comment by JFernando122** [comment]
ok, thanks, also while translating i found this (line 199)

![image](https://user-images.githubusercontent.com/40414805/122491012-c7e41100-cfa8-11eb-90d5-b288e1050c94.png)

I´m guessing it's supposed to be library

**Comment by subdiox** [comment]
`Book N of TITLE` (ex: `Book 5 of Kimetsu no Yaiba`) cannot be translated to Japanese by replacing the words "Book" and "of" respectively, because it is translated to "TITLE N巻" in Japanese (the order of the words are reversed). Is it possible to concatenate the words and translate the sentence like `Book %1 of %2`?

**Comment by OzzieIsaacs** [comment]
@subdiox: I changed it, the whole text is now translatable
@JFernando122: I'm not sure if I gave you feedback: It's corrected

**Comment by xfox26** [comment]
[messages.po.txt](https://github.com/janeczku/calibre-web/files/9136524/messages.po.txt)

Updated pt_BR

**Comment by OzzieIsaacs** [comment]
@xfox26: Thanks for the update. I think I have merged it and it is published in the latest version (0.6.19) of calibre-web

**Comment by arihid** [comment]
Here is my latest commit, using the recent language file.
[revision01.messages.po.txt](https://github.com/janeczku/calibre-web/files/10056023/revision01.messages.po.txt)

**Comment by hexin-lin-1024** [comment]
@OzzieIsaacs 
Hi, I have updated the Simplified Chinese translation to fix some poor translation and here it is [messages.po.txt](https://github.com/janeczku/calibre-web/files/10100263/messages.po.txt).
If you want to see further what has changed, please take a look at https://github.com/hexin-lin-1024/calibre-web.

**Comment by OzzieIsaacs** [comment]
@arihid 
> But I have a question, what is Plural-Forms?

Calibre-Web doesn't use this feature, so don't worry. I merged your translation into the newest nightly version of Calibre-Web

**Comment by arihid** [comment]
> @arihid
> 
> > But I have a question, what is Plural-Forms?
> 
> Calibre-Web doesn't use this feature, so don't worry. I merged your translation into the newest nightly version of Calibre-Web

Great! I will proof read it somewhere in the future.

**Comment by arihid** [comment]
Hi @janeczku, I just reviewed and proofreaded my last TL, and made some changes.
#2668 for details.

**Comment by UFervor** [comment]
@OzzieIsaacs Another Chinese translation update. 
[messages.po.txt](https://github.com/janeczku/calibre-web/files/10607816/messages.po.txt)
BTW, should fuzzy tags be removed after proofreading?

**Comment by OzzieIsaacs** [comment]
@UFervor: I will merge the changes next week. Please remove the fuzzy tags, because otherwise the texts are not used

**Comment by UFervor** [comment]
> @UFervor: I will merge the changes next week. Please remove the fuzzy tags, because otherwise the texts are not used

Fuzzy tags removed
[messages.po.txt](https://github.com/janeczku/calibre-web/files/10637199/messages.po.txt)

**Comment by OzzieIsaacs** [comment]
@UFervor It's merged and published. Thanks for the work

**Comment by UsamaFoad** [comment]
@OzzieIsaacs, I added the Arabic translation and created PR for it #3411

**Comment by emes81** [comment]
I'd be happy to help by adding en-GB to the project. It would be very little work, just minor changes to the en-US.
My main motivation for doing this would be to have an interface without the dreaded MM DD YYYY date format; is that part of the localisation or is that controlled elsewhere?
I'm a language worker and not a coder, any brief pointers on _how_ I go about adding en-GB would be appreciated.

**Comment by arihid** [comment]
> I'd be happy to help by adding en-GB to the project. It would be very little work, just minor changes to the en-US. My main motivation for doing this would be to have an interface without the dreaded MM DD YYYY date format; is that part of the localisation or is that controlled elsewhere? I'm a language worker and not a coder, any brief pointers on _how_ I go about adding en-GB would be appreciated.

For pointers, it's on the first post.
Or if you don't know what to do with that pointers, here's what you need to do:
Just download and edit 'messages.pot' from this project's root directory with any text editor and post it here. @janeczku should merge it for you. It's based on en_US so you just need to tweak some settings.

**Comment by Dunhill69** [comment]
updated Spanish translation 

[messages.po.txt](https://github.com/user-attachments/files/24994275/messages.po.txt)
